src/reducers/index.js

```diff
@@ -283,29 +284,33 @@ function initializeRowHeightsAndOffsets(state) {
 }
 
 function initializeFixedColumnWidthsAndOffsets(state) {
-  let { columnSettings } = state;
+  const { columnsCount, getColumn } = state.columnSettings;
   let fixedColumnsWidth = 0;
   let fixedRightColumnsWidth = 0;
-  const fixedColumns = new Array(columnSettings.fixedColumnsCount);
-  const fixedRightColumns = new Array(columnSettings.fixedRightColumnsCount);
-  const fixedColumnGroups = [];
-  const fixedRightColumnGroups = [];
-
-  for (let idx = 0; idx < fixedColumns.length; idx++) {
-    fixedColumns[idx] = convertColumnElementsToData(
-      columnSettings.getFixedColumn(idx)
-    );
-    fixedColumns[idx].props.index = idx;
-    fixedColumnsWidth += fixedColumns[idx].props.width;
+  const fixedColumns = [];
+  const fixedRightColumns = [];
+
+  for (let idx = 0; idx < columnsCount; idx++) {
+    const columnData = convertColumnElementsToData(getColumn(idx));
+    if (!columnData.props.fixed) {
+      break;
+    }
+    columnData.props.index = idx;
+    fixedColumnsWidth += columnData.props.width;
+    fixedColumns.push(columnData);
   }
 
-  for (let idx = 0; idx < fixedRightColumns.length; idx++) {
-    fixedRightColumns[idx] = convertColumnElementsToData(
-      columnSettings.getFixedRightColumn(idx)
-    );
-    fixedRightColumns[idx].props.index = idx;
-    fixedRightColumnsWidth += fixedRightColumns[idx].props.width;
+  for (let idx = columnsCount - 1; idx >= 0; idx--) {
+    const columnData = convertColumnElementsToData(getColumn(idx));
+    if (!columnData.props.fixedRight) {
+      break;
+    }
+    columnData.props.index = idx;
+    fixedRightColumnsWidth += columnData.props.width;
+    fixedRightColumns.push(columnData);
   }
+  fixedRightColumns.reverse();
```

Comment:
You can avoid reversing the list if you add elements at the start of the list. 
See [`unshift`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/unshift).